railties/lib/rails/command/base.rb

```diff
@@ -0,0 +1,135 @@
+require "thor"
+require "erb"
+
+require "active_support/core_ext/string/filters"
+require "active_support/core_ext/string/inflections"
+
+require "rails/command/actions"
+
+module Rails
+  module Command
+    class Base < Thor
+      class Error < Thor::Error # :nodoc:
+      end
+
+      include Actions
+
+      class << self
+        # Returns true when the app is a Rails engine.
+        def engine?
+          defined?(ENGINE_ROOT)
+        end
+
+        # Tries to get the description from a USAGE file one folder above the command
+        # root.
+        def desc(usage = nil, description = nil)
```

Comment:
Did you call this `desc` instead of `description` because someone else defines description already? Abbreviated this way I read it was `descending`.